src/main/java/dao/UserDaoImpl.java

```diff
@@ -0,0 +1,49 @@
+package dao;
+
+import exceptions.DataProcessingException;
+import java.util.List;
+import models.User;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class UserDaoImpl implements UserDao {
+    private final SessionFactory sessionFactory;
+
+    @Autowired
```

Comment:
Before Spring 4.3, we had to add an @Autowired annotation to the constructor. With newer versions, this is optional if the class has only one constructor.

Reply:
it looks like magic. )))